### DISCUSSION TITLE
[Feature Request] Ability to remove "All entries" from left panel of Entries page

### DISCUSSION THREAD

**Discussion by benface** [craftcms/cms#8537] [Ideas]
For most of my sites, it doesn't make sense to mix all the different sections and entry types in the same list. I would love a simple option to toggle the visibility of the "All entries" source without using a plugin. I imagine it would go in the "Customize sources" modal.

<img width="235" alt="all-entries" src="https://user-images.githubusercontent.com/1059139/39094054-d947455e-45f7-11e8-8d4b-60c4e772c6d8.png">

**Comment by brandonkelly** [comment]
Related: #1630

**Comment by benface** [comment]
If anyone is looking to do this now, I was able to do it with a few lines in a module:

```php
// Remove "All Entries" source
Event::on(
    Entry::class,
    Entry::EVENT_REGISTER_SOURCES,
    function (Event $event) {
        array_shift($event->sources);
    }
);
```

**Comment by BenParizek** [comment]
I'm (probably) not against this in some cases, but my main concern with this would be: _How do I just search across everything if I don't know where something is?_

I recently inherited a site that was poorly architected and the developer had done this (hidden the "All Entries" option) using custom CSS in the Control Panel. It was not clear at all where certain entries were and the client was complaining about a poor experience finding what they were looking for as they had to click on each Section to search for what they wanted (they did not know they could remove the CSS hiding the "All Entries" option).

For your consideration.
